File: Array/leetcode/easy/MaximumConsecutiveOnes.py
def find_max_consecutive_ones(arr):
    max_count = 0
    ans = 0
    
    for num in arr:
        if num == 1:
            max_count += 1
        else:
            ans = max(ans, max_count)
            max_count = 0

    ans = max(ans, max_count)
    return ans

# Driver function
def find_max_consecutive_ones_main():
    arr = [1, 1, 0, 1, 1, 1]
    print(find_max_consecutive_ones(arr))  # Output: 3

find_max_consecutive_ones_main()

File: Array/leetcode/easy/MaximumConsecutiveOnes.go
package main

import "fmt"

func findMaxConsecutiveOnes(arr []int) int {
	maxCount := 0
	ans := 0
	for i := 0; i < len(arr); i++ {
		if arr[i] == 1 {
			maxCount++
		} else {
			ans = max(ans, maxCount)
			maxCount = 0
		}
	}
	ans = max(ans, maxCount)
	return ans
}
func findMaxConsecutiveOnesMain() {
	arr := []int{1, 1, 0, 1, 1, 1}
	fmt.Println(findMaxConsecutiveOnes(arr)) // Output: 3
}
